distributed/scheduler.py

```diff
@@ -987,24 +977,13 @@ def duration(self) -> float:
         """The total amount of time spent on all tasks belonging to this collection"""
         return self._duration_us / 1e6
 
-    def transition(self, old: TaskStateState, new: TaskStateState) -> None:
-        self.states[old] -= 1
-        self.states[new] += 1
-
     @property
     def types(self) -> Set[str]:
         """The result types of this collection"""
         return self._types.keys()
 
-    def update_nbytes(self, diff: int) -> None:
-        self.nbytes_total += diff
-
-    @staticmethod
-    def _calculate_duration_us(start: float, stop: float) -> int:
-        return max(round((stop - start) * 1e6), 0)
-
     def __len__(self) -> int:
-        return sum(self.states.values())
+        return self._size
```

Comment:
I've seen plugins accessing this and this can also be costly if calculated on demand